.github/workflows/dart-tests.yaml

```diff
@@ -0,0 +1,90 @@
+name: Relic CI
+
+on:
+  push:
+    branches:
+      - main
+      - dev
+      - add-dart-project-and-ci-setup # temporary, testing
+  pull_request:
+    branches:
+      - main
+      - dev
+
+env:
+  DART_VERSION: '3.3.0'
+  PUB_CACHE_PATH: ~/.pub-cache
+
+jobs:
+  dart_format:
+    name: Verify Dart Formatting
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [windows-latest, ubuntu-latest]
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v3
+      - name: Setup Dart
+        uses: dart-lang/setup-dart@v1.3
+        with:
+          sdk: ${{ env.DART_VERSION }}
+      - name: Cache Dart dependencies
+        uses: actions/cache@v3
+        with:
+          path: ${{ env.PUB_CACHE_PATH }}
+          key: ${{ runner.os }}-pub-cache-${{ env.DART_VERSION }}
+          restore-keys: |
+            ${{ runner.os }}-pub-cache-
+      - name: Verify formatting
+        run: dart format --output=none --set-exit-if-changed .
+
+  dart_analyze:
```

Comment:
Suggestion: Add downgrade tests for the analyzer as well. (this can be observed in the Serverpod repo).